### PR TITLE
test: reach 100% apiparser.py coverage

### PR DESCRIPTION
**Pull request by kayl-codes**
## Proposed change
Extends the apiparser.py test suite (merged in #29) to 100% statement coverage and adds `pytest-cov` as a dev dependency/tooling for measuring it.

## Type of change
- [x] Code quality improvements to existing code or addition of tests

## Additional information
No production code changes — test-only PR.

## Checklist
- [x] The code change is tested and works locally.
- [x] The code has been formatted and linted using Ruff.
- [x] Tests have been added to verify that the new code works.
- [ ] Documentation added/updated if required.
- [x] Tested against the latest Home Assistant version.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Extend API parser test suite to fully cover additional edge cases and behaviors while adding coverage tooling for local measurement.

Enhancements:
- Add tests for from_entry, generate_keymap, matches_only, fill_defaults, parse_api, fill_ensure_vals, and fill_vals_proc to cover previously untested branches and edge conditions.

Build:
- Add pytest-cov as a development dependency for coverage measurement.

Tests:
- Expand apiparser.py tests to achieve 100% statement coverage, including conversions, default handling, and value processing paths.

### CODE REVIEW

File: tests/test_apiparser.py [hunk 1]
```diff
@@ -328,6 +369,90 @@ def test_parse_api_no_key_no_search_single_dict_targets_root(ap: ModuleType) ->
     assert result == {"total": 42}
 
 
+def test_parse_api_entry_without_matching_uid_is_skipped(ap: ModuleType) -> None:
+    """An entry lacking ``key``/``key_secondary``/``key_search`` is dropped."""
+    source = [{"id": "1", "name": "pool0"}, {"name": "orphan"}]
+    result = ap.parse_api(source=source, key="id", vals=[{"name": "name"}])
+    assert result == {"1": {"name": "pool0"}}
+
+
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply tests/test_apiparser.py:378]: Fixed in 6997bdb — added `test_parse_api_all_entries_without_matching_uid_returns_empty` covering the case where every entry lacks a matching key.

**Comment by kayl-codes** [comment]
@sourcery-ai review
